felix/labelindex/named_port_index.go

```diff
@@ -422,30 +502,37 @@ func (idx *SelectorAndNamedPortIndex) UpdateIPSet(ipSetID string, sel selector.S
 		// which isn't currently possible in Felix, since the ID is formed by hashing the other
 		// values.  For completeness, handle (inefficiently) by simulating a deletion.
 		log.WithField("ipSetID", ipSetID).Warn("IP set selector or named port changed for existing ID.")
+		for m := range oldIPSetData.memberToRefCount {
+			// Emit deletion events for the members.  We don't need to do that
+			// for a real IP set deletion because it's handled en-masse.
+			idx.OnMemberRemoved(ipSetID, m)
```

Comment:
New UTs found a latent bug here, but it's on a defensive code path so not sure it's worth backporting.  At least it's now tested!